src/testFixtures/java/org/opensearch/knn/KNNRestTestCase.java

```diff
@@ -173,17 +187,85 @@ public void cleanUpCache() throws Exception {
         clearCache();
     }
 
-    /**
-     * Set up cluster settings for remote index build feature. We do this for all tests to ensure the fallback mechanisms are working correctly.
-     */
     @Before
     public void setupRemoteIndexBuildSettings() throws Exception {
-        if (randomBoolean() && isRemoteIndexBuildSupported(getBWCVersion())) {
+        final String remoteBuild = System.getProperty("test.remoteBuild", null);
+        if (isRemoteIndexBuildSupported(getBWCVersion()) && remoteBuild != null) {
             updateClusterSettings(KNNFeatureFlags.KNN_REMOTE_VECTOR_BUILD_SETTING.getKey(), true);
             updateClusterSettings(KNNSettings.KNN_REMOTE_VECTOR_REPO, "integ-test-repo");
+            updateClusterSettings(KNNSettings.KNN_REMOTE_BUILD_SERVICE_ENDPOINT, "http://0.0.0.0:80");
+            updateClusterSettings(KNNSettings.KNN_REMOTE_BUILD_CLIENT_POLL_INTERVAL, TimeValue.timeValueSeconds(0));
+            setupRepository("integ-test-repo");
+            BEFORE_INDEX_BUILD_SUCCESS_COUNT = getRemoteIndexBuildSuccessCount();
+        } else if (isRemoteIndexBuildSupported(getBWCVersion()) && randomBoolean()) {
+            // Set up cluster settings for remote index build feature. We do this for all tests to ensure the fallback mechanisms are
+            // working correctly.
+            updateClusterSettings(KNNFeatureFlags.KNN_REMOTE_VECTOR_BUILD_SETTING.getKey(), true);
+            updateClusterSettings(KNNSettings.KNN_REMOTE_VECTOR_REPO, "integ-test-repo");
+        }
+    }
+
+    @Retention(RetentionPolicy.RUNTIME)
+    @Target(ElementType.METHOD)
+    public @interface ExpectRemoteBuildValidation {
+    }
```

Comment:
Loved this. Thanks @anntians for contributing the annotation this should really help ease out the development.